### PR TITLE
Add system information to UserAgent in AccountManager requests

### PR DESCRIPTION
**Pull request by huffman**

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Build console output for commit: 3e18f064b835ad37da9b22b119c4516b91b52660 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by huffman** [comment]
To test:

 * Smoke test. There's nothing specific to test functionality-wise.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0dbb3f0f7f220507b35803798a12b94b79d584f9 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by huffman** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0dbb3f0f7f220507b35803798a12b94b79d584f9 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by sethalves** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 0dbb3f0f7f220507b35803798a12b94b79d584f9 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: e5bcc44aa92ffa1f6619987eca92e50d4c530918 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by huffman** [comment]
please test this

**Comment by huffman** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: e5bcc44aa92ffa1f6619987eca92e50d4c530918 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by ZappoMan** [comment]
@huffman - your PR has gotten old and stale and needs to be rebased.

**Comment by huffman** [comment]
Rebased

**Comment by hifi-gustavo** [comment]
Build console output for commit: 14a388fe63b29231c00c64d392cf992728b8185d available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by huffman** [comment]
To test:

 * Test that the client and server connect and work together
 * Test that creating a temporary place name works via the Sandbox "Share" button

File: interface/src/scripting/AccountScriptingInterface.cpp [hunk 1]
```diff
@@ -13,20 +13,22 @@
 
 #include "AccountScriptingInterface.h"
 
+#include <DependencyManager.h>
```
Comment (Atlante45) [inline interface/src/scripting/AccountScriptingInterface.cpp]: You should never have to include DependencyManager except in the dependency header.

File: libraries/networking/src/AccountManager.h [hunk 2]
```diff
@@ -49,10 +52,12 @@ Q_DECLARE_METATYPE(AccountManagerAuth::Type);
...
 const QByteArray ACCESS_TOKEN_AUTHORIZATION_HEADER = "Authorization";
 
-class AccountManager : public QObject {
+using UserAgentGetter = std::function<QString()>;
+
+class AccountManager : public QObject, public Dependency {
     Q_OBJECT
 public:
-    static AccountManager& getInstance(bool forceReset = false);
+    AccountManager(UserAgentGetter = []() -> QString { return HIGH_FIDELITY_USER_AGENT; });
```
Comment (Atlante45) [inline libraries/networking/src/AccountManager.h]: variable name missing.
Would probably be cleaner to have a default like:
```cpp
UserAgentGetter DEFAULT_USER_AGENT_GETTER = []{ return HIGH_FIDELITY_USER_AGENT; }
```

Comment (huffman) [inline libraries/networking/src/AccountManager.h]: Good call, fixed

File: interface/src/scripting/AccountScriptingInterface.cpp (hunk 1 above)
Comment (huffman) [inline interface/src/scripting/AccountScriptingInterface.cpp]: Hmm, not sure why I added this. Removed.

**Comment by hifi-gustavo** [comment]
Build console output for commit: c7d31f1d17d308f88b1f3bc7530f46ed3ce6b922 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: ee15a76c3b9b0c219c094f54101edd7ea354c095 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by Atlante45** [comment]
@huffman You need to update.

**Comment by hifi-gustavo** [comment]
Build console output for commit: 64c89a5dde45d83fb785ce40d8e66b07bc69a204 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by huffman** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 64c89a5dde45d83fb785ce40d8e66b07bc69a204 available at: https://gist.github.com/3684e783aca7e97319ed4e7b51169667<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR7834.dmg

**Comment by OmegaHeron** [comment]
QA Pass
Testing Windows 10 64 bit - i5 6500 CPU - GTX970 GPU - **Desktop mode - keyboard/mouse**
Test that the client and server connect and work together - Yes
Test that creating a temporary place name works via the Sandbox "Share" button - Yes
Icon for face capture and mic is back in lower right corner - explained by comments in #7908

All standard tests pass.  This PR to latest RC and DEV stacks is good.

**Pull request merged**